Src/Test.UnitTests.Sarif.PatternMatcher/PatternInvariantTests.cs

```diff
@@ -60,42 +63,90 @@ public static void VerifyAllValidatorsExist(string definitionsFilePath)
                         string.Join($",{Environment.NewLine}  ", rulesWithoutValidators));
         }
 
-        public static void VerifyAllRuleFilenamesMatchDefinitions(string definitionsFilePath)
+        /// <summary>
+        /// This function ensures that for each shared strings variable definition, 
+        /// there is a corresponding regex call to it in the JSON.
+        /// </summary>
+        public static void VerifyAllJsonRulesExist(string definitionsFilePath, string sharedStringsFilePath)
+        {
+            // This function verifies that for each shared string variable definition, there is a rule in the JSON that uses it
+            string definitionsFileContents = File.ReadAllText(definitionsFilePath);
+            string sharedStringsContents = File.ReadAllText(sharedStringsFilePath);
+
+            string line;
+            var reader = new StringReader(sharedStringsContents);
+
+            var sharedStringsWithoutRules = new List<string>();
+            while ((line = reader.ReadLine()) != null)
+            {
+                line = line.Trim();
+                if (!line.StartsWith('$'))
+                {
+                    continue;
+                }
+
+                line = line.Split('=')[0];
+                if (!definitionsFileContents.Contains(line))
+                {
+                    sharedStringsWithoutRules.Add(line);
+                }
+            }
+
+            // Assert.Empty doesn't allow custom messages, so use Assert.True
+            Assert.True(sharedStringsWithoutRules.Count == 0,
+                        "Found no reference to these regular expression definitions in JSON: " +
+                        $"{Environment.NewLine}  " +
+                        string.Join($",{Environment.NewLine}  ", sharedStringsWithoutRules));
+
+        }
+
+        /// <summary>
+        /// This function ensures that for each SEC101_... filename, there is a corresponding rule name and ID match in the JSON.
+        /// </summary>
+        public static void VerifyAllRuleFilenamesMatchDefinitions(string definitionsFilePath, string validatorsFolderName)
         {
             //Read the json file content to get the rule names
             string content = File.ReadAllText(definitionsFilePath);
             SearchDefinitions sdObject = JsonConvert.DeserializeObject<SearchDefinitions>(content);
-            var regexSet = new HashSet<string>();
+            var nameIdDictionary = new Dictionary<string, string>();
+            var invalidFilenames = new List<string>();
+
             foreach (SearchDefinition searchDefinition in sdObject.Definitions)
             {
                 foreach (MatchExpression matchExpression in searchDefinition.MatchExpressions)
                 {
-                    regexSet.Add(matchExpression.Name.Split('/')[1]);
+                    string ruleName = matchExpression.Name.Split('/')[1];
+                    string ruleID = matchExpression.Id.Replace('/', '_');
```

Comment:
>string ruleID = matchExpression.Id.Replace('/', '_'); [](http://example.com/codeflow?start=20&length=53)

The correct formatting for a ruleId includes the forward slash (`/`). I take it replace is to match how ruleIds are used in file names?

Reply:
Exactly, I ensure the filenames are the same by checking if the filename starts with this edited ruleID

Reply:
I did however remove this logic from the test that ensures rules only have 1 ID assigned to it since the filename checking doesn't happen in that function.